tests/test_conda_meta_json_filename.py

```diff
@@ -0,0 +1,53 @@
+"""
+Test that wheel files installed via conda-pypi create JSON files in conda-meta/
+with the correct filename format: name-version-build.json
+"""
+
+from __future__ import annotations
+
+import json
+from pathlib import Path
+
+
+def test_extract_whl_sets_fn_correctly(
+    pypi_demo_package_wheel_path: Path,
+    tmp_path: Path,
+):
+    """
+    Test that extract_whl_as_conda_pkg sets the fn field correctly in index.json.
+    This is a unit test that directly tests the metadata creation.
+    """
+    from conda_pypi.pre_command.extract_whl import extract_whl_as_conda_pkg
+
+    extract_whl_as_conda_pkg(str(pypi_demo_package_wheel_path), str(tmp_path))
+
+    # Check that index.json was created with correct fn field
+    index_json_path = tmp_path / "info" / "index.json"
+    assert index_json_path.exists(), f"index.json should exist at {index_json_path}"
```

Comment:
Is this a common pattern? Adding f strings in assert statements? What's the purpose of this? 🤔